### ISSUE TITLE
AnalysisLevel property does not affect Code Style

### ISSUE THREAD

**Issue by dupuyjs** [dotnet/roslyn#63036]
### Describe the bug
Based on the documentation of [AnalysisLevel](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#analysislevel) property, this property is supposed to affect also Code Style starting with .NET 6.

>Starting in .NET 6, if you set [EnforceCodeStyleInBuild](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild) to true, this property affects [code-style (IDEXXXX) rules](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/) too.

However, IDEXXXX warnings are not displayed, except if you configure all categories independently.

### To Reproduce

When using `AnalysisLevel` with `EnforceCodeStyleInBuild`, only Code Analysis issues are reported, not Code Style. (82 warnings in my project, none related to IDEXXXX rules)

```
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <AnalysisLevel>latest-All</AnalysisLevel>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
```

If I use the following configuration instead, by setting all categories individually, both Code Quality and Code Style issues are correctly reported. (150 warnings in my project). Note: `AnalysisLevel` is not referenced in this configuration. 

```
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <AnalysisLevelDesign>latest-All</AnalysisLevelDesign>
    <AnalysisLevelDocumentation>latest-All</AnalysisLevelDocumentation> 
    <AnalysisLevelGlobalization>latest-All</AnalysisLevelGlobalization>
    <AnalysisLevelInteroperability>latest-All</AnalysisLevelInteroperability> 
    <AnalysisLevelMaintainability>latest-All</AnalysisLevelMaintainability>
    <AnalysisLevelNaming>latest-All</AnalysisLevelNaming>
    <AnalysisLevelPerformance>latest-All</AnalysisLevelPerformance> 
    <AnalysisLevelSingleFile>latest-All</AnalysisLevelSingleFile> 
    <AnalysisLevelReliability>latest-All</AnalysisLevelReliability> 
    <AnalysisLevelSecurity>latest-All</AnalysisLevelSecurity>
    <AnalysisLevelUsage>latest-All</AnalysisLevelUsage>
    <AnalysisLevelStyle>latest-All</AnalysisLevelStyle>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
```

And if I add `AnalysisLevelStyle` in addition to `AnalysisLevel`, Code Style issues are still not reported (still 82 warnings).

```
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <AnalysisLevel>latest-All</AnalysisLevel>
    <AnalysisLevelStyle>latest-All</AnalysisLevelStyle>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
```

### Further technical details

``` bash
.NET SDK (reflecting any global.json):
 Version:   6.0.302
 Commit:    c857713418

Runtime Environment:
 OS Name:     debian
 OS Version:  11
 OS Platform: Linux
 RID:         debian.11-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.302/

global.json file:
  Not found

Host:
  Version:      6.0.7
  Architecture: x64
  Commit:       0ec02c8c96

.NET SDKs installed:
  6.0.302 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.7 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

Visual Studio Code

``` bash
Version: 1.69.2 (system setup)
Commit: 3b889b090b5ad5793f524b5d1d39fda662b96a2a
Date: 2022-07-18T16:12:52.460Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Windows_NT x64 10.0.22000
```

**Comment by mawosoft** [comment]
Just ran into this as well. The problem is here:
https://github.com/dotnet/roslyn/blob/87b5be1d9314398c2db4029580281d62be71fdba/src/CodeStyle/Tools/Program.cs#L244-L248

As @freljung observed, the snipped explicitly checks that `AnalysisLevel` and `AnalysisLevelStyle` (or the `Mode` properties) are not the same. Not sure why, this seems to be the initial code from 3 years ago that never got looked at again. I assume whatever reason may have existed for that comparison at the time no longer applies.

Also hard to find in this repo because the the targets file doesn't exist anywhere and is instead generated by the little program above.

**Comment by palapapa** [comment]
I just ran into this issue as well.

**Comment by anthonymastreanvae** [comment]
Any idea in which SDK version this fix was released?

**Comment by JoeRobich** [comment]
@anthonymastreanvae

It was part of the 9.0.100 release and caused build failures for every project that had set TreatWarningsAsErrors because they were not expecting code style warnings to be emitted during their build. We quickly backed the change out in 9.0.101. I do not expect CodeStyle analysis to ever be tied to the AnalysisLevel property.

Instead you should be able to set `<AnalysisLevelStyle>`.

**Comment by anthonymastreanvae** [comment]
Should we be able to at least _spell_ the category names properly in .NET 10?

```xml
  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
    <AnalysisLevel>latest-All</AnalysisLevel>
    <AnalysisLevelStyle>latest-All</AnalysisLevelStyle>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>
```

Will this workaround no longer be necessary?

https://github.com/dotnet/roslyn/issues/63036#issuecomment-1483923853

**Comment by JoeRobich** [comment]
> Should we be able to at least spell the category names properly in .NET 10?

I don't understand this question.

**Comment by anthonymastreanvae** [comment]
The two earlier comments suggested that the problem occurred when the levels were set to the same values OR when the uppercase category name was used, requiring us to use lowercase and to arbitrarily differentiate the values.

https://github.com/dotnet/roslyn/issues/63036#issuecomment-1326373960
https://github.com/dotnet/roslyn/issues/63036#issuecomment-1483923853
